### PR TITLE
update large image cell and web view cell

### PR DESCRIPTION
**Pull request by ZevEisenberg**
Also add some extensions to clean up configuration boilerplate code.

### CODE REVIEW

File: Auto Layout/CollectionViewController.swift [hunk 1]
```diff
@@ -85,39 +86,51 @@ final private class CollectionViewDataSource: NSObject, UICollectionViewDataSour
...
+        kicker: "KICKER",
+        credit: "Photo by Joe Blow",
+        hideFooter: false,
+        crop: Crop(imageFilename: "coltrane.jpg", size: CGSize(width: 540, height: 300))
+      )
+      return cell
+    case .webCellSection:
+      let cell: WebCell = collectionView.dequeue(for: indexPath)
+      cell.set(
+        url: URL(string: "https://secure-ds.serving-sys.com/BurstingRes/Site-85296/WSFolders/7649898/TH029_728x90_r3.hyperesources/TH029_728x90_GiGi.jpg")!,
```
Comment (wsu-nyt) [inline Auto Layout/CollectionViewController.swift:130]: For demo purposes, I think the `!` here might be ok.

File: Auto Layout/Configurable.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+protocol Configurable {}
+
+extension Configurable {
+  func configure(_ closure: (inout Self) -> Void) -> Self {
+    var mutable = self
+    closure(&mutable)
+    return mutable
+  }
+}
```
Comment (wsu-nyt) [inline Auto Layout/Configurable.swift:19]: 👏 Nice!

File: Auto Layout/LargeImageCell.swift [hunk 3]
```diff
@@ -10,83 +10,99 @@ import UIKit
...
+        kickerLabel,
+        headlineLabel
+      ]),
+      summaryLabel,
+      footerView,
+    ])
+    contentView.addSubview(mainStack)
+    mainStack.pinEdgesToSuperView(lowerBottomAndTrailingPriorities: true)
+    mainStack.isLayoutMarginsRelativeArrangement = true
+    mainStack.layoutMargins = UIEdgeInsets(all: 10)
```
Comment (wsu-nyt) [inline Auto Layout/LargeImageCell.swift:65]: The usage of `UIStackView` makes total sense here. 👍

**Review by wsu-nyt — APPROVED** [review]
Really nice work!

**Review by JonLz — APPROVED** [review]

File: Podfile [hunk 4]
```diff
@@ -6,3 +6,6 @@ target 'Sample' do
   pod 'Texture', '~> 3.0'
   pod 'PINRemoteImage', :git => 'https://github.com/pinterest/PINRemoteImage', :branch => 'master'
 end
+
+target 'SampleUITests' do
+end
```
Comment (wsu-nyt) [inline Podfile:11]: I just realized that this new target is added without any dependencies. but I guess this is expected?

Comment (ZevEisenberg) [reply Podfile:11]: Jon found that we had to add it to get the tests to build correctly…or something. I forget the details, but at minimum, it shouldn't break anything.

File: Sample/Section.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
+//
+//  Section.swift
+//  Sample
+//
+//  Created by Zev Eisenberg on 9/10/20.
+//  Copyright © 2020 The New York Times. All rights reserved.
+//
+
+enum Section: CaseIterable {
```
Comment (ZevEisenberg) [inline Sample/Section.swift:9]: We added this to standardize the section logic between the two demos. Previously, the Texture example was making a peculiar use of modular arithmetic to make a quasi-FizzBuzz, pseudorandom distribution of cells that was probably slightly more realistic, but also made testing very confusing.

**Pull request merged**